server/models/LegacyTree.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * This is a temporary model that allows to replicate data from field data schema to the
+ * treetracker database for backward compatability while the breakdown/migration of monolith
+ * db is in process. It can be removed at appropriate time when the dependencies on
+ * legacy tables in treetracker main db is bound to its relevant contexts. For e.g, tree_attributes
+ * is really capture_metadata and has to be owned by a future capture service context and the current
+ * trees table in treetracker db is really a tree capture table and do not represent a unique tree..
+ */
+const { Repository } = require('./Repository');
+const LegacyTree = ({ id, image_url, lat, lon, planter_id, planter_identifier }) => Object.freeze({
+    uuid: id,
+    image_url,
+    lat,
+    lon,
+    planter_id,
+    planter_identifier,
+    time_created: new Date().toISOString(),
+    time_updated: new Date()
+});
+const createTreesInMainDB = (legacyTreeRepoImpl, legacyTreeAttrRepoImpl) => (async (tree, attributes) => {
+    const legacyTreeRepository = new Repository(legacyTreeRepoImpl);
+    const legacyAttributesRepository = new Repository(legacyTreeAttrRepoImpl);
+    const result = await legacyTreeRepository.save(tree);
+    const tree_attributes = attributes.map(attribute =>  Object.assign({ tree_id: result.id }, attribute));
+    legacyAttributesRepository.save(tree_attributes);

```

Comment:
Is it okay to ignore the `await` for this promise?

Reply:
My reasoning for not adding await here is because I only care about persisting `tree_attributes` and not rely on values from resolving the promise. But I may not be handling a exception if it were to occur here, I am afraid.  The calling code handles the exception if it were to be thrown and rolls back the transaction. But I believe I should catch the exception and rethrow it like below
``legacyAttributesRepository.save(tree_attribues).catch(err => throw err);``
Am I right?

Reply:
I will trigger some error conditions and see the behavior and update the code as well.